### DISCUSSION TITLE
Best way to determine if price is trending or rangebound and find breakouts?

### DISCUSSION THREAD

**Discussion by preintel** [DaveSkender/Stock.Indicators#629] [Q&A]
Wanted to see how other users are determining if current price channels are trending or rangebound and best way to find breakouts in price. 

I figure a number of coders will end up using ATR (average trading range) with lookback to determine ranges of price action or develop ways to determine support/resistance points to determine breakouts. Alternatively have also seen the use of Bollinger bands/standard deviations from price normal distributions to attempt to determine break outs. Have seen the use of higher highs and lower lows to find momentum and the use of turning points to determine price reversion.

Are there any specific indicators or mathematical models you prefer to use to find these points and determine the trend. Any combination of methods to lessen noise without losing breakout points?

**Comment by DaveSkender** [comment]
I don’t have a specific answer to your question, but have always thought about candlestick patterns (#220) when thinking about breakouts, so it might be best solved by a combination at factors and pattern recognition.  Investor Business Daily has always taken [this approach](https://www.investors.com/ibd-university/how-to-buy/common-patterns-1/) to determine base formations and breakouts.
